ibis/backends/sql/compilers/base.py

```diff
@@ -51,15 +52,7 @@ def AlterTable(*args, kind="TABLE", **kwargs):
     from ibis.backends.sql.datatypes import SqlglotType
 
 
-def get_leaf_classes(op):
-    for child_class in op.__subclasses__():
-        if not child_class.__subclasses__():
-            yield child_class
-        else:
-            yield from get_leaf_classes(child_class)
-
-
-ALL_OPERATIONS = frozenset(get_leaf_classes(ops.Node))
+ALL_OPERATIONS = frozenset(get_subclasses(ops.Node))
```

Comment:
These aren't equivalent. Before we were getting only the leaf classes, now we're getting all classes (including any intermediate base classes like `Binary`). Was that intentional?

Reply:
Hm, good point. I suppose this implementation needs to be adjusted, or I can add a flag to `get_subclasses`

Reply:
Looking into how this is used, I actually think it's better this way. We have some ops that are subclasses of other (non base-class/abstract) ops. Having some extraneous methods like `visit_Binary` seems less bad than accidentally excluding things like `Median`.

Reply:
Hm, also a good point, I'll revert